src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/ReduceTokensCodeCleanupProvider.vb

```diff
@@ -88,7 +88,7 @@ Namespace Microsoft.CodeAnalysis.CodeCleanup.Providers
 
                         Dim base = literal.GetBase()
 
-                        If Not base.HasValue Then
+                        If Not base.HasValue OrElse idText.Contains("_") Then
```

Comment:
Add a comment explaining what's going on here.  If someone unfamiliar with VB or digit separators reads this code, they might not know what's happening.